packages/nuxt/src/app/composables/ssr.ts

```diff
@@ -14,7 +14,11 @@ export function useRequestHeaders (include?: any[]) {
   const event = useRequestEvent()
   const headers = event ? getRequestHeaders(event) : {}
   if (!include) { return headers }
-  return Object.fromEntries(include.map(key => key.toLowerCase()).filter(key => headers[key]).map(key => [key, headers[key]]))
+  return Object.fromEntries(include.map(key => {
+   // Testing codspeed 
+    const lowerKey = key.toLowerCase()
+    return [lowerKey, headers[lowerKey]]
+  }).filter(([_, value]) => value !== undefined))
```

Comment:
I think this iterate approach would be more performant still.

```suggestion
  const headers = Object.create(null)
  for (const _key of include) {
    const key = _key.toLowerCase()
    const header = headers[key]
    if (header) {
      headers[key] = header
    }
  }
```

Reply:
I still have a question about why this sort of filtering at the end is needed.. It will just return undefined either way.

Reply:
It creates a smaller object

Reply:
I see. I'm gonna run some benchmarks for this

Reply:
It only iterates once instead of 3 times in your example, and 4 in the original code.

Reply:
@danielroe I made some benchmarks, and Object.create(null) is ~10x slower than {}. Is there a reason we want to avoid the built-in properties for the headers if it's not used outside of the function, meaning for everything else it will be a readonly table?
![image](https://github.com/nuxt/nuxt/assets/92037085/0b9711f3-9609-4042-9e36-f47a052a415a)

Reply:
it's to avoid prototype pollution. admittedly i can't think of an exploit as the headers should all be _strings_ but it's safer when assigning user-controlled keys to an object. And it's only initialised 1x per `useRequestHeaders()` call.

Reply:
Well, yea, I guess in such case this bit of performance can be sacrificed in favor of security.

Reply:
Is this good? Should we maybe use the headers returned from getRequestHeaders?